core/templates/pages/profile-page/profile-page.component.ts

```diff
@@ -96,10 +103,10 @@ angular.module('oppia').component('profilePage', {
             }
           );
 
-          ctrl.userNotLoggedIn = !data.username;
+          ctrl.userNotLoggedIn = !data.currentUser;
```

Comment:
This seems like a functionality change. Where is currentUser defined?

If it is an error, why was it not caught?

Reply:
(Note: this is probably covered by the comment below.)

Reply:
Yes it's defined in `user-profile-object.factory.ts`.